wandb/sync/sync.py

```diff
@@ -158,6 +165,10 @@ def _send_tensorboard(self, tb_root, tb_logdirs, send_manager):
         # send all of our records like a boss
         while not send_manager._interface.record_q.empty():
             data = send_manager._interface.record_q.get(block=True)
+            if len(data.history.ListFields()) != 0:
+                item = data.history.item.add()
+                item.key = "_step"
+                item.value_json = json.dumps(data.history.step.num)
```

Comment:
This worries me too.  I had thought setting `publish_step` would do the trick but we actually have to handle setting step in multiple places?

Reply:
HandlerThread does a similar logic in _history_assign_step. This just aims to replicate that using the step given from the logs